client/src/main/java/io/pravega/client/admin/impl/StreamManagerImpl.java

```diff
@@ -184,6 +196,31 @@ public boolean deleteScope(String scopeName, boolean deleteStreams) throws Delet
                     throw new DeleteScopeFailedException(message, e);
                 }
             }
+
+            Iterator<KeyValueTableInfo> kvtIterator = controller.listKeyValueTables(scopeName).asIterator();
+            while (kvtIterator.hasNext()) {
+                KeyValueTableInfo kvt = kvtIterator.next();
+                try {
+                    Futures.getThrowingException(controller.deleteKeyValueTable(scopeName, kvt.getKeyValueTableName()));
+                } catch (Exception e) {
+                    String message = String.format("Failed to delete key-value table %s", kvt.getKeyValueTableName());
+                    throw new DeleteScopeFailedException(message, e);
+                }
+            }
+
+            for (String groupName: readerGroupList) {
+                try {
+                    Futures.getThrowingException(controller.getReaderGroupConfig(scopeName, groupName)
+                            .thenCompose(conf -> controller.deleteReaderGroup(scopeName, groupName,
```

Comment:
remind me again, this will delete the RG stream as well, right?

Reply:
Yes. However, the RG stream is already being deleted early on while deleting all the streams. So the `deleteStream` call during `deleteReaderGroup` is idempotent.